ha/setup/ha_setup.py

```diff
@@ -196,31 +198,84 @@ def process(self):
-                Log.info(f"Put node in standby output: {standby_output}")
-                if json.loads(standby_output).get("status") != STATUSES.FAILED.value:
-                    Log.info("Creating pacemaker resources")
-                    create_all_resources(s3_instances=s3_instances)
-                    Log.info("Created pacemaker resources successfully")
+        self._cluster_manager = CortxClusterManager()
+        # Check if the cluster exists already, if not, create one.
+        cluster_exists = self._confstore.key_exists(const.CLUSTER_CONFSTORE_NODES_KEY)
+        Log.info(f"Cluster exists? {cluster_exists}")
+        if cluster_exists == False:
+            try:
+                # Create cluster
+                Log.info(f"Creating cluster: {cluster_name} with node: {node_name}")
+                cluster_output: str = self._cluster_manager.cluster_controller.create_cluster(
+                    cluster_name, cluster_user, cluster_secret, node_name)
+                Log.info(f"Cluster creation output: {cluster_output}")
+                if json.loads(cluster_output).get("status") == STATUSES.SUCCEEDED.value:
+                    # Put cluster in standby mode
+                    standby_output: str = self._cluster_manager.node_controller.standby(node_name)
+                    Log.info(f"Put node in standby output: {standby_output}")
+                    if json.loads(standby_output).get("status") != STATUSES.FAILED.value:
+                        Log.info("Creating pacemaker resources")
+                        create_all_resources(s3_instances=s3_instances)
+                        Log.info("Created pacemaker resources successfully")
+                        # Add this node to the cluster nodes list in the store.
+                        self._confstore.set(f"{const.CLUSTER_CONFSTORE_NODES_KEY}/{node_name}")
+                    else:
+                        raise HaConfigException(f"Failed to put cluster in standby mode. Error: {standby_output}")
                 else:
-                    raise HaConfigException(f"Failed to put cluster in standby mode. Error: {standby_output}")
+                    raise HaConfigException(f"Cluster creation failed. Error: {cluster_output}")
+            except Exception as e:
+                Log.error(f"Cluster creation failed; destroying the cluster. Error: {e}")
+                output = self._execute.run_cmd(const.PCS_CLUSTER_DESTROY, check_error=True)
+                Log.info(f"Cluster destroyed. Output: {output}")
+                # Delete the node from nodelist if it was added in the store
+                if self._confstore.key_exists(f"{const.CLUSTER_CONFSTORE_NODES_KEY}/{node_name}"):
+                    self._confstore.delete(f"{const.CLUSTER_CONFSTORE_NODES_KEY}/{node_name}")
+                raise HaConfigException("Cluster creation failed")
+        else:
+            node_added = False
+            # If this node in cluster already.
```

Comment:
Put in log
`# If this {node} node in cluster already`

Reply:
done